supervisor/core.py

```diff
@@ -310,6 +317,48 @@ def _update_last_boot(self):
         self.sys_config.last_boot = self.sys_hardware.helper.last_boot
         self.sys_config.save_data()
 
+    async def _adjust_system_datetime(self):
+        """Adjust system time/date on startup."""
+        # If no timezone is detect or set
+        # If we are not connected or time sync
+        if (
+            self.sys_config.timezone
+            or self.sys_host.info.timezone not in ("Etc/UTC", None)
+        ) and (self.sys_host.info.dt_synchronized or self.sys_supervisor.connectivity):
+            return
+
+        # Get Timezone data
+        try:
+            data = await retrieve_whoami(self.sys_websession)
```

Comment:
So it's not 100% clear why we would auto detect it here. Why can't we rely on the timezone from HA?

Reply:
We do, but at this point, we have maybe no core and no user settings and need to provide a working system before the user can make any configuration. Also, timezone changes on containers are a bit static, we need to create the system close as possible to the user timezone. In the end, it's like the timezone setting on core, it's only a high-level application setting and has only an effect on the functions we care about, all other C-libs call using the timezone from the container. All work only if all parts are correct setup, and the container is already exists at the point which user can change the timezone.